test/minikube/minikube_domain_resync_test.go

```diff
@@ -101,6 +101,55 @@ func checkArchives(t *testing.T, namespaceName string, adminPod string, numExpec
 	return
 }
 
+func TestAdminScaleDown(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{
+		SetValues: map[string]string{
+			"admin.replicas": "2",
+		},
+	}, 2, "")
+
+	adminStatefulSet := helmChartReleaseName + "-nuodb-cluster0"
+	admin0 := adminStatefulSet + "-0"
+	admin1 := adminStatefulSet + "-1"
+
+	// scale down Admin StatefulSet
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+	k8s.RunKubectl(t, options, "scale", "statefulset", adminStatefulSet, "--replicas=1")
+
+	// wait for scaled-down Admin to show as "Disconnected"
+	testlib.Await(t, func() bool {
+		output, _ := k8s.RunKubectlAndGetOutputE(t, options, "exec", admin0, "--",
+			"nuocmd", "show", "domain", "--server-format", "{id} {connected_state}")
+		return strings.Contains(output, admin1+" Disconnected")
+	}, 300*time.Second)
+
+	// wait for scaled-down Admin Pod to be deleted
+	testlib.AwaitNoPods(t, namespaceName, admin1)
+
+	// commit a Raft command to confirm that remaining Admin has consensus
+	k8s.RunKubectl(t, options, "exec", admin0, "--",
+		"nuocmd", "set", "value", "--key", "testKey", "--value", "testValue", "--unconditional")
```

Comment:
this should not be the case without the K8s aware module. What is the guarantee that the module processed the leaving admin in time? Would this not be flaky?

Reply:
The conditions for a scaled-down Admin being excluded from consensus are that (1) the StatefulSet replica count is lower than or equal to the ordinal in the server ID and (2) the Pod with that server ID (the server ID and Pod name should be identical for this behavior to be enabled) is deleted. We checked earlier in the test that Pod was deleted. Even if the surviving Admin did not receive the "Pod deleted" event by the time we invoke `nuocmd set value`, all blocking Raft commands are issued with a timeout (the defaults vary, but a typical timeout is 5 seconds. So eventually the surviving member would notice that it has consensus to commit the command and would notify the client that the command is committed.

Reply:
can you actively check whether a server is excluded from membership? Does it show in "show domain" or any other REST API call?